src/Microsoft.ML.Featurizers/DateTimeTransformer.cs

```diff
@@ -0,0 +1,782 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Runtime.InteropServices;
+using System.Security;
+using System.Text;
+using Microsoft.ML;
+using Microsoft.ML.CommandLine;
+using Microsoft.ML.Data;
+using Microsoft.ML.EntryPoints;
+using Microsoft.ML.Featurizers;
+using Microsoft.ML.Internal.Utilities;
+using Microsoft.ML.Runtime;
+using Microsoft.ML.Transforms;
+using Microsoft.Win32.SafeHandles;
+using static Microsoft.ML.Featurizers.CommonExtensions;
+
+[assembly: LoadableClass(typeof(DateTimeTransformer), null, typeof(SignatureLoadModel),
+    DateTimeTransformer.UserName, DateTimeTransformer.LoaderSignature)]
+
+[assembly: LoadableClass(typeof(IRowMapper), typeof(DateTimeTransformer), null, typeof(SignatureLoadRowMapper),
+   DateTimeTransformer.UserName, DateTimeTransformer.LoaderSignature)]
+
+[assembly: EntryPointModule(typeof(DateTimeTransformerEntrypoint))]
+
+namespace Microsoft.ML.Featurizers
+{
+
+    public static class DateTimeTransformerExtensionClass
+    {
+        /// <summary>
+        /// Create a <see cref="DateTimeTransformerEstimator"/>, which splits up the input column specified by <paramref name="inputColumnName"/>
+        /// into all its individual datetime components. Input column must be of type Int64 representing the number of seconds since the unix epoc.
+        /// This transformer will append the <paramref name="columnPrefix"/> to all the output columns. If <paramref name="columnsToDrop"/> is empty,
+        /// then all the columns are returned. Otherwise, the columns listed in the array will be dropped from the return value.
+        /// </summary>
+        /// <param name="catalog">Transform catalog</param>
+        /// <param name="inputColumnName">Input column name</param>
```

Comment:
Is it possible to name the output columns? ML.NET typically allows the user to name the output column.

Reply:
This Transformer returns 21 different columns. Instead of specifying an output name for each column, the  columns already have a predefined name (Year, Month, Date, Etc), and the user can just specify a prefix to add to each column.

---
In reply to: [353910044](https://github.com/dotnet/machinelearning/pull/4521#discussion_r353910044) [](ancestors = 353910044)